File: update_rag_api.py
import re

with open('lib/services/rag_api_service.dart', 'r') as f:
    content = f.read()

content = content.replace("import 'database_service.dart';", "import 'database_service.dart';\nimport 'ai_backend.dart';")
content = re.sub(r"static const String _baseUrl =\s*'https://rag-backend-647731796550.asia-south1.run.app';", "", content)
content = re.sub(r"static Map<String, String> _headers\(\) \{[\s\S]*?\}", "", content)

content = content.replace("Uri.parse('$_baseUrl/api/chat')", "Uri.parse('${AiBackend.baseUrl}/api/chat')")
content = content.replace("Uri.parse('$_baseUrl/api/chat/stream')", "Uri.parse('${AiBackend.baseUrl}/api/chat/stream')")
content = content.replace("Uri.parse('$_baseUrl/api/inventory/sync')", "Uri.parse('${AiBackend.baseUrl}/api/inventory/sync')")
content = content.replace("Uri.parse('$_baseUrl/api/cache/clear')", "Uri.parse('${AiBackend.baseUrl}/api/cache/clear')")

content = content.replace("_headers()", "AiBackend.headers()")

with open('lib/services/rag_api_service.dart', 'w') as f:
    f.write(content)

File: update_rag_api.dart
import 'dart:io';

void main() {
  final file = File('lib/services/rag_api_service.dart');
  var content = file.readAsStringSync();
  
  content = content.replaceFirst(
    "import 'database_service.dart';",
    "import 'database_service.dart';\nimport 'ai_backend.dart';"
  );
  
  content = content.replaceAll(
    "static const String _baseUrl =\n      'https://rag-backend-647731796550.asia-south1.run.app';",
    ""
  );

  content = content.replaceAll(
    "static Map<String, String> _headers() {\n    final companyId = DatabaseService().companyId;\n    return {\n      'Content-Type': 'application/json',\n      if (companyId.isNotEmpty) 'x-company-id': companyId,\n    };\n  }",
    ""
  );

  content = content.replaceAll("Uri.parse('\$_baseUrl/api/chat')", "Uri.parse('\${AiBackend.baseUrl}/api/chat')");
  content = content.replaceAll("Uri.parse('\$_baseUrl/api/chat/stream')", "Uri.parse('\${AiBackend.baseUrl}/api/chat/stream')");
  content = content.replaceAll("Uri.parse('\$_baseUrl/api/inventory/sync')", "Uri.parse('\${AiBackend.baseUrl}/api/inventory/sync')");
  content = content.replaceAll("Uri.parse('\$_baseUrl/api/cache/clear')", "Uri.parse('\${AiBackend.baseUrl}/api/cache/clear')");
  
  content = content.replaceAll("_headers()", "AiBackend.headers()");

  file.writeAsStringSync(content);
}
